dev-infrastructure/modules/aks-cluster-base.bicep

```diff
@@ -469,6 +469,36 @@ resource uami_fedcred 'Microsoft.ManagedIdentity/userAssignedIdentities/federate
   }
 ]
 
+resource pullerIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' = {
+  location: location
+  name: 'image-puller'
+}
+
+module acrPullerRoles 'acr/acr-permissions.bicep' = [
+  for (_, i) in acrPullResourceGroups: {
+    name: guid(acrRg[i].id, aksCluster.id, acrPullRoleDefinitionId)
+    scope: acrRg[i]
+    params: {
+      principalId: pullerIdentity.properties.principalId
+      acrResourceGroupid: acrRg[i].id
+    }
+  }
+]
+
+resource puller_fedcred 'Microsoft.ManagedIdentity/userAssignedIdentities/federatedIdentityCredentials@2023-01-31' = [
+  for i in range(0, length(workloadIdentities)): {
+    parent: uami[i]
+    name: '${workloadIdentities[i].value.uamiName}-${location}-puller-fedcred'
+    properties: {
+      audiences: [
+        'api://AzureCRTokenExchange'
+      ]
+      issuer: aksCluster.properties.oidcIssuerProfile.issuerURL
+      subject: 'system:serviceaccount:${workloadIdentities[i].value.namespace}:${workloadIdentities[i].value.serviceAccountName}'
```

Comment:
after running this template i get the error `Issuer and subject combination already exists for this Managed Identity`

so an MI can be federated only once with a certain SA on a cluster BUT the audience list can only hold one value.

https://learn.microsoft.com/en-us/entra/workload-id/workload-identity-federation-considerations